### ISSUE TITLE
CPU usage of qbittorrent

### ISSUE THREAD

**Issue by ixzh** [ClementTsang/bottom#495]
## Describe the bug

Apprarently my system was freezing...and I checked qbittorrent's CPU usage, where I found bottom shows different CPU% from htop.

## How to reproduce
![image](https://user-images.githubusercontent.com/14981695/121838490-8df5df00-cd0a-11eb-9808-0721d4650d6a.png)

## bottom version
0.6.1

## bottom config file (if used)

nil

## Installation method

AUR

## Platform and environment information
Arch
### Operating system, OS version, and architecture
```
OS: ArcoLinux
Kernel: 5.12.10-arch1-1
Uptime: 16 hours
Packages: 1263 (pacman)
Shell: bash 5.1.8
Resolution: 1920x1080
WM: LG3D
Theme: Arc-Dark [GTK2/3]
Icons: Sardi-Arc [GTK2/3]
Terminal: alacritty
CPU: Intel i5-4200U (4) @ 2.600GHz
GPU: Intel Haswell-ULT
Memory: 1905MiB / 3834MiB (49%)
_SWAP size:_ 2GB
```

**Comment by ClementTsang** [comment]
Sorry if this sounds a bit incoherent, I just woke up.

TL;DR: This isn't a bug, this is intentional behaviour because htop and bottom are calculating entirely different things.  htop by default on that column does *not* divide over the number of cores, bottom does.

---
I'm like 99% sure this is just a case htop not dividing by cores in that column while bottom does.

For example, if I look at CPU% now at htop for something like rust analyzer, it shows over 100% usage:
![image](https://user-images.githubusercontent.com/34804052/121841828-1fceff00-ccad-11eb-8808-7f5a15a50826.png)

Meanwhile, if I look at bottom, it's far lower, only saying something like 6.2%, which makes a bit more sense:
![image](https://user-images.githubusercontent.com/34804052/121841853-32493880-ccad-11eb-9cd2-bc830360fa96.png)

This 6.2% value is also what GNOME system monitor says if I check, so I'm clearly not the only one calculating like this:

![image](https://user-images.githubusercontent.com/34804052/121841879-3e34fa80-ccad-11eb-8ac8-44124aed3113.png)

But is htop wrong?  Or is bottom?  Neither, actually, they're *both* right - because they're calculating and showing different things!

bottom uses a normalized CPU usage, affected by the number of cores.  Meanwhile, htop defaults to not doing this.  We can test this hypothesis pretty easily - I'm on a 16 core system, the usage I'm seeing on htop is about 100%, 100 / 16 is 6.25 - and I was seeing about 6.2%, wasn't I?

Or if we take your example, you saw about 46% usage on htop, you have 4 cores... 46 / 4 is about 11.5% usage, which is what bottom shows!

But we don't need sketchy napkin math to reach this conclusion, as this different behaviour is baked right into htop if you want, through their "NCPU%" or "PERCENT_NORM_CPU" column!

![image](https://user-images.githubusercontent.com/34804052/121842272-1eea9d00-ccae-11eb-8bc2-f613908808a7.png)

As for *why* bottom defaults to showing normalized CPU usage, it's because most of the time, IMO, it makes more sense for a general user.  Adding the option to support non-normalized usage would be nice I guess, but it's a bit of a low priority action to tackle.

**Comment by ixzh** [comment]
> 41% divided over 4 cores is _pretty_ close to what bottom saw.

Yeah, i roughly sense that. Thank you for the detailed explanation.  But also it make some sense 他哦show overall load on all cores/threads, as the compiuter did slow down significantly and I can't pin down qbittorrent in this scenario?

**Comment by ClementTsang** [comment]
Yeah, adding an overall non-normalized column can be helpful and I totally should do it, it's just that my plate is currently very full in terms of both IRL stuff and bottom.

Wouldn't be hard to do either, just... currently don't have the time.  It would also be blocked on me implementing customizable columns, which is *further* blocked by some current work I'm doing.

So... uh, hopefully soon, haha.

**Comment by ClementTsang** [comment]
Opened and issue for the suggested feature in #496

**Comment by ClementTsang** [comment]
Closing since this seems to be resolved.
